compiler/rustc_mir/src/borrow_check/diagnostics/mod.rs

```diff
@@ -388,7 +388,7 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                     //
                     // We know the field exists so it's safe to call operator[] and `unwrap` here.
                     let (&var_id, _) =
-                        self.infcx.tcx.typeck(def_id.expect_local()).closure_captures[&def_id]
+                        self.infcx.tcx.typeck(def_id.expect_local()).closure_min_captures[&def_id]
                             .get_index(field.index())
```

Comment:
I guess the question still stands here, I assume that we want to instead get the var_hir_id of the root variable instead of the one that we are getting right now.

Reply:
@logmosier the above refers to this thing. We don't want to get the root variable, we want to get the place and then do get root_variable.
as it stands this won't work if u have something like

If `p.x`, `p.y`,  `c` were captured for some p and c. Then index 2 will be out of bounds when we try refer to `c` here because captures look like: closure_min_captures[closure_def_id] = { p -> [Place(projs:Field(x)), Place(projs:Field(y))], c -> [Place()] }